declarative-config-bridge/src/main/java/io/opentelemetry/instrumentation/config/bridge/ConfigPropertiesBackedDeclarativeConfigProperties.java

```diff
@@ -0,0 +1,207 @@
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Implementation of {@link DeclarativeConfigProperties} backed by {@link ConfigProperties}.
+ *
+ * <p>It tracks the navigation path and only resolves to system properties at the leaf node when a
+ * value is actually requested.
+ */
+public final class ConfigPropertiesBackedDeclarativeConfigProperties
+    implements DeclarativeConfigProperties {
+
+  private static final String GENERAL_PEER_SERVICE_MAPPING = "general.peer.service_mapping";
+
+  private static final Map<String, String> LIST_MAPPINGS;
+
+  static {
+    LIST_MAPPINGS = new HashMap<>();
+    LIST_MAPPINGS.put(
+        "general.http.client.request_captured_headers",
+        "otel.instrumentation.http.client.capture-request-headers");
+    LIST_MAPPINGS.put(
+        "general.http.client.response_captured_headers",
+        "otel.instrumentation.http.client.capture-response-headers");
+    LIST_MAPPINGS.put(
+        "general.http.server.request_captured_headers",
+        "otel.instrumentation.http.server.capture-request-headers");
+    LIST_MAPPINGS.put(
+        "general.http.server.response_captured_headers",
+        "otel.instrumentation.http.server.capture-response-headers");
+  }
+
+  private final ConfigProperties configProperties;
+  private final List<String> path;
+
+  public static DeclarativeConfigProperties createInstrumentationConfig(
+      ConfigProperties configProperties) {
+    return new ConfigPropertiesBackedDeclarativeConfigProperties(
+        configProperties, Collections.emptyList());
+  }
+
+  private ConfigPropertiesBackedDeclarativeConfigProperties(
+      ConfigProperties configProperties, List<String> path) {
+    this.configProperties = configProperties;
+    this.path = path;
+  }
+
+  @Nullable
+  @Override
+  public String getString(String name) {
+    String fullPath = pathWithName(name);
+    return configProperties.getString(toPropertyKey(fullPath));
+  }
+
+  @Nullable
+  @Override
+  public Boolean getBoolean(String name) {
+    String fullPath = pathWithName(name);
+    return configProperties.getBoolean(toPropertyKey(fullPath));
+  }
+
+  @Nullable
+  @Override
+  public Integer getInt(String name) {
+    String fullPath = pathWithName(name);
+    return configProperties.getInt(toPropertyKey(fullPath));
+  }
+
+  @Nullable
+  @Override
+  public Long getLong(String name) {
+    String fullPath = pathWithName(name);
+    return configProperties.getLong(toPropertyKey(fullPath));
+  }
+
+  @Nullable
+  @Override
+  public Double getDouble(String name) {
+    String fullPath = pathWithName(name);
+    return configProperties.getDouble(toPropertyKey(fullPath));
+  }
+
+  /**
+   * Important: this method should return null if there is no structured child with the given name,
+   * but unfortunately that is not implementable on top of ConfigProperties.
+   *
+   * <p>This will be misleading if anyone is comparing the return value to null.
+   */
```

Comment:
this isn't ideal, but I think we can live with it

Reply:
I'm hoping to just use the Declarative Configuration API directly without any additional checks, but this is a good option to consider if this becomes an issue